File: atm cajero sockets/cajero-cliente-go/src/atm/client/socket_process.go
// Equivalente a atm/client/socket_process.py: en Java es una interface, en
// Python una ABC; en Go usamos directamente una interface del lenguaje.
package client

type SocketProcess interface {
	Connect() bool
	Listen() []interface{}
	Response(data []interface{}) bool
	Close() bool
}

File: atm cajero sockets/cajero-cliente/src/atm/client/socket_process.py
"""
Equivalente a lenin.client.SocketProcess del profesor:
en Java es una interface, en Python usamos ABC.
"""

from abc import ABC, abstractmethod
from typing import List


class SocketProcess(ABC):

    @abstractmethod
    def connect(self) -> bool:
        raise NotImplementedError

    @abstractmethod
    def listen(self) -> List[object]:
        raise NotImplementedError

    @abstractmethod
    def response(self, data: List[object]) -> bool:
        raise NotImplementedError

    @abstractmethod
    def close(self) -> bool:
        raise NotImplementedError
